src/rgw/rgw_common.h

```diff
@@ -358,7 +358,7 @@ class RGWHTTPArgs {
   bool exist_obj_excl_sub_resource() const {
     const char* const obj_sub_resource[] = {"append", "torrent", "uploadId",
                                             "partNumber", "versionId"};
-    for (int i = 0; i != std::size(obj_sub_resource); i++) {
+    for (unsigned i = 0; i != std::size(obj_sub_resource); i++) {
```

Comment:
this change is also included as https://github.com/ceph/ceph/pull/39998/commits/e777c1046b68ebe1debea2c2073b3cd93a7190fd .